gui/webdisplay/src/TWebWindowsManager.cxx

```diff
@@ -593,29 +617,58 @@ bool ROOT::Experimental::TWebWindowsManager::Show(ROOT::Experimental::TWebWindow
       is_firefox = prog.Length() > 0;
 
 #ifdef _MSC_VER
-      if (win.IsBatchMode())
+      if (batch_mode)
          // there is a problem when specifying the window size with wmic on windows:
          // It gives: Invalid format. Hint: <paramlist> = <param> [, <paramlist>].
-         exec = gEnv->GetValue("WebGui.FirefoxBatch", "fork: -headless -no-remote $url");
+         exec = gEnv->GetValue("WebGui.FirefoxBatch", "fork: -headless -no-remote $profile $url");
       else
-         exec = gEnv->GetValue("WebGui.FirefoxInteractive", "$prog -width=$width -height=$height $url");
+         exec = gEnv->GetValue("WebGui.FirefoxInteractive", "$prog -width=$width -height=$height $profile $url");
 #else
-      if (win.IsBatchMode())
-         exec = gEnv->GetValue("WebGui.FirefoxBatch", "fork:-headless -no-remote $url");
+      if (batch_mode)
+         exec = gEnv->GetValue("WebGui.FirefoxBatch", "fork:-headless -no-remote $profile $url");
       else
-         exec = gEnv->GetValue("WebGui.FirefoxInteractive", "$prog -width $width -height $height \'$url\' &");
+         exec = gEnv->GetValue("WebGui.FirefoxInteractive", "$prog -width $width -height $height $profile \'$url\' &");
 #endif
+
+      if (is_firefox && (exec.Index("$profile") != kNPOS)) {
+         TString repl;
+
+         const char *ff_profile = gEnv->GetValue("WebGui.FirefoxProfile","");
+         const char *ff_profilepath = gEnv->GetValue("WebGui.FirefoxProfilePath","");
+         Int_t ff_randomprofile = gEnv->GetValue("WebGui.FirefoxRandomProfile", 0);
+         if (ff_profile && *ff_profile) {
+            repl.Form("-P %s", ff_profile);
+         } else if (ff_profilepath && *ff_profilepath) {
+            repl.Form("-profile %s", ff_profilepath);
+         } else if ((ff_randomprofile > 0) || (batch_mode && (ff_randomprofile>=0))) {
+
+            gRandom->SetSeed(0);
+
+            TString ffp, ffd;
```

Comment:
Can you use longer variable names that tell us what "ffp" "ffd" contain?

Reply:
ok :)